### ISSUE TITLE
[Bug]: Unable to tab to the month and year navigation controls in the Datepicker Component

### ISSUE THREAD

**Issue by kvaidyanathan-ns1** [carbon-design-system/carbon#18395]
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.72.0

### Description

Unable to tab to the month and year navigation controls in the Datepicker Component

https://github.com/user-attachments/assets/6f41cb0d-f937-450b-a1cd-c438452ca178

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-datepicker--overview#range-datepicker

### Steps to reproduce

- Tab through the controls in the datepicker component.
- Select a start date and attempt to change the year and month controls.
- Similarly, for the end date, try changing the month and year controls.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
